### ISSUE TITLE
Collection<...> generates invalid code

### ISSUE THREAD

**Issue by Joao-Neves** [google/FreeBuilder#227]
From version 1.10.6 onwards every time I write something like:

```Java
@FreeBuilder
public interface MyBuilder {

    Collection<MyClass> getMyClass();

    class Builder extends MyBuilder_Builder {}
} 
```

I get the following compilation error:

> Error:(xxx, xx) java: incompatible types: java.util.Collection<mypackage.MyClass> cannot be converted to java.util.List<mypackage.MyClass>

By inspecting the generated code we can see that this occurs when you attempt to do this `myClass = value.getMyClass();` but the field `myClass ` is defined as `List<MyClass>` instead of  `Collection<MyClass>`.

**Comment by chrisalice** [comment]
Thanks for the spot! As a workaround you can use List instead of Collection until we get this fixed, if binary backwards compatibility is not an issue for you.

**Comment by chrisalice** [comment]
Note that this bug also affects wildcard types like `List<? extends Number>`.

**Comment by chrisalice** [comment]
@Joao-Neves Could you give v1.12.1-rc1 a try?

**Comment by Joao-Neves** [comment]
@chrisalice Sure thing, I'll test it as soon as I get to the office tomorrow morning.

**Comment by Joao-Neves** [comment]
@chrisalice Hello, it's working fine on my end. Thank you for the quick fix!
Happy Holidays!

**Comment by chrisalice** [comment]
Happy holidays!

**Comment by chrisalice** [comment]
This should be fixed in [v1.12.1](https://github.com/google/FreeBuilder/releases/tag/v1.12.1) proper now.
